### ISSUE TITLE
Misleading error when providing malformed auth credentials

### ISSUE THREAD

**Issue by halkyon** [storj/gateway-mt#198]
When authenticating with Gateway-MT, if a client provides credentials that are malformed or invalid, the error returned from Gateway-MT can be misleading. For example, if you provide an incorrect date in the authorization header, it instead returns an error saying no access key could be found.

We have a credential middleware that parses auth given, and fetches access grant from the authservice. It checks for an error but doesn't do anything useful with it: https://github.com/storj/gateway-mt/blob/main/pkg/server/middleware/credentials.go#L52. If minio was able to parse the credential due to a bug in our parser, it then goes to https://github.com/storj/gateway-mt/blob/main/pkg/minio/iam-storj-store.go#L70 it can't find them from the context, so returns a not found error.

An option is to add more validation, and change our credential parsing middleware from "speculative" to erroring completely if it can't parse. I believe we could have the middleware write an appropriate S3 response if there's an error, such as https://github.com/storj/minio/blob/main/cmd/api-errors.go#L685, but there's also a range of credential errors we could use as well, making it easier to identify a buggy client.

**Comment by shaupt131** [comment]
Added to edge team backlog.  Thanks Sean :)

**Comment by halkyon** [comment]
I started changing our credential parsing to not be speculative, and try to detect the signature type and return an error if it can't parse: https://review.dev.storj.io/c/storj/gateway-mt/+/8048. This is also how minio works. Currently if it can't parse, we get a whole stack of errors for all types it couldn't parse, so it's not possible to give a good error response on what exactly happened. The hardest part will be making sure this doesn't break anything.

I explored another option: falling back to what minio parsed if we can't parse the credentials ourselves: https://review.dev.storj.io/c/storj/gateway-mt/+/8067.
